### ISSUE TITLE
Support for kubernetes 1.25

### ISSUE THREAD

**Issue by astylos** [apache/pulsar-helm-chart#289]
I am unable to upgrade Pulsar version after upgrading kubernetes to 1.25.0

```
$ microk8s kubectl get nodes
NAME          STATUS   ROLES    AGE    VERSION
centos8vm02   Ready    <none>   348d   v1.25.0
centos8vm03   Ready    <none>   348d   v1.25.0
centos8vm01   Ready    <none>   348d   v1.25.0
centos8vm04   Ready    <none>   348d   v1.25.0
```

```
$ microk8s.helm3 --namespace pulsar upgrade pulsar-mini apache/pulsar --version 2.9.4 -f pulsar.yaml
Error: UPGRADE FAILED: [resource mapping not found for name: "pulsar-mini-bookie" namespace: "pulsar" from "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
ensure CRDs are installed first, resource mapping not found for name: "pulsar-mini-broker" namespace: "pulsar" from "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
ensure CRDs are installed first, resource mapping not found for name: "pulsar-mini-proxy" namespace: "pulsar" from "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
ensure CRDs are installed first, resource mapping not found for name: "pulsar-mini-zookeeper" namespace: "pulsar" from "": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
ensure CRDs are installed first]
```

I'm no kubernetes/helm expert, so may have done this incorrectly, but I pulled the chart from git, edited all `policy/v1beta1` to `policy/v1` and tried installing from local chart, but it still gave same errors.

**Comment by relief-melone** [comment]
Fix should be straight forward. As workaround you can just disable the policy by

```yaml
bookie:
  pdb:
    usePolicy: false
```

In the chart as PodDisruptionBudget has been moved out of beta and has been stable since v1.21 the chart should look something like this

```yaml
{{- if semverCompare ">=1.21" .Capabilities.KubeVersion.GitVersion }}
apiVersion: policy/v1
{{- else }}
apiVersion: policy/v1beta1
{{- end }}
kind: PodDisruptionBudget
```

**Comment by michaeljmarshall** [comment]
This will be fixed by #260.
